### PR TITLE
Added manual and changed modules to use STG instead of concat

### PR DESCRIPTION
**Pull request by jrbeaumont**

### CODE REVIEW

File: doc/manual.md [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+use an example and go through it based on lines, explaining what is
+necessary. For this example we will be using the same as example as in
+previous sections, from the examples included with the concepts repo.
+This is “*Celement_with_env_1.hs*”.
+
+Concepts file layout
+--------------------
+
+![A concepts file](http://jrbeaumont.github.io/concepts-manual/images/concepts_file_screenshot.svg)
+
```
Comment (snowleopard) [inline doc/manual.md:370]: I suggest to simply use a Haskell code snippet here, e.g.:

``` haskell
module Concept where

import Tuura.Concept.Concat
...
```

A screenshop from Sublime Text editor may be confusing (readers might think it's required to use concepts).

**Comment by snowleopard** [comment]
Looks very nice. Just one comment above.

**Comment by jrbeaumont** [comment]
@snowleopard: As you'll see I've included all the movement of modules in this.
I also need to make some changes to Workcraft for this, as it has a default emtpy concepts file using the wrong module import

**Comment by snowleopard** [comment]
Great! You'll also need to generate new screenshots, as currently they still show `Concat`.

**Comment by snowleopard** [comment]
@jrbeaumont Let me know when you are done with the screenshots, and I'll merge this.

**Comment by jrbeaumont** [comment]
Oh right, that's a problem.

I'll do it when I get home, I'm currently on OS X and screenshotting is a pain.

**Comment by snowleopard** [comment]
No problem, take your time.

**Comment by jrbeaumont** [comment]
@snowleopard: The images have been updated.

**Comment by snowleopard** [comment]
OK, let's now drop type signatures `Eq a => a -> ... -> CircuitConcept a` from the docs.

**Comment by jrbeaumont** [comment]
@snowleopard: Type signatures dropped. All tested and working. Manual updated.

**Comment by snowleopard** [comment]
@jrbeaumont Awesome! Merged now.

**Pull request merged**